### PR TITLE
bpo-35606: Fix math.prod tests using 'start' as keyword parameter

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue35606

### CODE REVIEW

File: Lib/test/test_math.py [hunk 1]
```diff
@@ -1803,14 +1803,14 @@ def test_prod(self):
...
-        self.assertRaises(TypeError, prod, values, bytearray(b''))
+        self.assertRaises(TypeError, prod, values, start=bytearray(b''))
         self.assertRaises(TypeError, prod, [[1], [2], [3]])
         self.assertRaises(TypeError, prod, [{2:3}])
-        self.assertRaises(TypeError, prod, [{2:3}]*2, {2:3})
-        self.assertRaises(TypeError, prod, [[1], [2], [3]], [])
+        self.assertRaises(TypeError, prod, [{2:3}]*2, start={2:3})
+        self.assertRaises(TypeError, prod, [[1], [2], [3]], start=[])
         with self.assertRaises(TypeError):
             prod([10, 20], [30, 40])     # start is a keyword-only argument
```
Comment (serhiy-storchaka) [inline Lib/test/test_math.py]: It looks not clear.
```suggestion
            prod([10, 20], 1)     # start is a keyword-only argument
```

Comment (serhiy-storchaka) [reply Lib/test/test_math.py]: You can add also tests for weird cases like `prod([2, 3], start='ab')`, `prod([2, 3], start=[1, 2])` and `prod([], start={2: 3})`.

Comment (pablogsal) [reply Lib/test/test_math.py]: > You can add also tests for weird cases like `prod([2, 3], start='ab')`, `prod([2, 3], start=[1, 2])` and `prod([], start={2: 3})`.

Done!

File: lel.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import traceback
```
Comment (vstinner) [inline lel.py]: Really? 🧐

Comment (pablogsal) [reply lel.py]: Ah, you discovered the secret file. You won 100000$ :)

File: Modules/mathmodule.c [hunk 3]
```diff
@@ -3082,14 +3082,13 @@ math_prod_impl(PyObject *module, PyObject *iterable, PyObject *start)
     }
 
     if (result == NULL) {
-        result = PyLong_FromLong(1);
+        result = _PyLong_GetOne();
         if (result == NULL) {
```
Comment (vstinner) [inline Modules/mathmodule.c]: _PyLong_GetOne() cannot be NULL.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**